content/techniques/logger.md

```diff
@@ -65,13 +81,36 @@ export class MyLogger extends Logger {
 }
 ```
 
+As shown in the **Custom logger** section above, pass an instance of your extended logger via the logger property of the options object in `NestFactory.create()`.
+
+#### Use built-in logger
+
+To use the built-in logger **in your own code**, simply instantiate the `Logger` in your class, and take advantage of its optional `context` constructor argument, which allows you to pass in a string to uniquely identify the logging context. The context string will automatically be appended to your log message:
+
+```typescript
+import { Injectable, Logger } from '@nestjs/common';
+
+@Injectable()
+export class CatsService {
+  private readonly cats: Cat[] = [];
+  private readonly logger = new Logger('CatsService');
```

Comment:
This partially solves the problem because it's still hard to mock them/disable logs when running, e.g. unit tests. In the best-case scenario, logger should be registered as a provider and injected through a constructor. We don't have to update the source code, but we should at least mention what would be the best to do here. :D

Reply:
Would it be suggested to create a `log` method in each class that uses the logger so it would be easy to add the context to each log call. Something like
```ts
@Injectable()
export class MyService {

  constructor(private readonly logger: Logger) {}

  private log(message: string, level: 'log' | 'error' | 'warn' | 'verbose' | 'debug') {
    switch(level) {
       case 'info': this.logger.log(message, MyService.name);
                    break;
       /* same follows for other cases with correct level */
    }
  }
}
```
Or is there some way to set the context so that it doesn't need to be sent each time? I know I solved this in my application with a TRANSIENT logger that could be injected with a dynamic module. Just wondering if there is another way you'd suggest